### ISSUE TITLE
Exec format error : Cannot start container <id> : exec format error 

### ISSUE THREAD

**Issue by lresende** [docker/docker#10668]
Build an image where you add a boostrap.sh starting with # which will be run when the container starts.

When docker run, it will produce the following error
exec format errorFATA[0000] Error response from daemon: Cannot start container d51c63c1ab2c9d8906aa7a6d427d9f41df8303ffe88b234bab82c9cb983f2f64: exec format error 

Dockerfile
```Dockerfile
FROM centos:6.6
MAINTAINER Luciano Resende
USER root
ADD bootstrap.sh /etc/bootstrap.sh
RUN chown root:root /etc/bootstrap.sh
RUN chmod 700 /etc/bootstrap.sh
ENV BOOTSTRAP /etc/bootstrap.sh
CMD ["/etc/bootstrap.sh", "-bash"]
```

Bootstrap.sh
```bash
#-------------------------------------------------------------
# xxxx
# xxxx
# xxxx
# xxxx
# xxxx
# xxx
#-------------------------------------------------------------

#!/usr/bin/env bash
if [[ $1 = "-d" ]]; then
  while true; do sleep 1000; done
fi
if [[ $1 = "-bash" ]]; then
  /bin/bash
fi
```

I tested by removing the comments from boostrap.sh and rebuilding it and then the image starts working ok again.

**Comment by phemmer** [comment]
This is expected. When running an executable, the kernel looks at the first few bytes to determine the type of the file. For scripts this means the first line needs to be a [shebang](https://en.wikipedia.org/wiki/Shebang_%28Unix%29). The shebang tells the kernel how to execute the script (that it needs to use an interpreter, and which interpreter to use).

**Comment by lresende** [comment]
Is this an error that is coming from Docker ? Could we at least have a better user error message where it make things easier to debug/solve the problem ?

**Comment by aleksey-korolev** [comment]
+1 for better erorr message

**Comment by Di3g0** [comment]
+1 for better error message

**Comment by thaJeztah** [comment]
This is an error message that's produced by the kernel, so if you want a better error message, please report at https://lkml.org

**Comment by jasetom** [comment]
Had the same issue, fixed it by adding `#!/bin/sh` at the top of the file instead of having other comments

**Comment by orian** [comment]
It may mean that the binary was build for system not compatible with docker base image system.

**Comment by GlenKPeterson** [comment]
@orian you mean like running a 64-bit image on a 32-bit host OS?  I suspect that's why I just got this message.

**Comment by orian** [comment]
@GlenKPeterson yes, exactly. It happen to me to compile the Go binary on Mac and try to run it on dockerize Ubuntu. The error is very helpful in such case.

**Comment by hefekranz** [comment]
#!/bin/sh in my entrypoint fixed it for me

**Comment by pcgeek86** [comment]
I encountered this error with Docker Compose. In order to fix it, I needed to:

- Add `#!/bin/sh` to the `main.sh` script, defined as the `ENTRYPOINT` in the `Dockerfile`
- Force rebuild of containers: `docker-compose up -d --build --force-recreate`

**Comment by Oomi** [comment]
I am facing the same issue with docker. How do I open this file Bootstrap.sh to edit. I am unable to find it. I have found Bootstrap.jar

**Comment by badal1443** [comment]
I am very new to Docker community, I started experimenting with docker on 1 of my Ubuntu 14.04 machine with i686 architecture, I see following error. 
"FATA[0000] Error response from daemon: Cannot start container 5db84c34e065162caadcd4bbcff67ff964ab30b6a7224d30a412c3a2228103ad: [8] System error: exec format error"

I googled and found that Docker doesn't work with Linux other than 64bit. Looks like I am getting _system error:exec format error_ because of it, can somebody comment on it?

Thanks!
